synapse/storage/relations.py

```diff
@@ -73,12 +162,29 @@ def get_relations_for_event(
             where_clause.append("type = ?")
             where_args.append(event_type)
 
-        order = "ASC"
+        if aggregation_key:
+            where_clause.append("aggregation_key = ?")
+            where_args.append(aggregation_key)
+
+        pagination_clause = generate_pagination_where_clause(
+            direction=direction,
+            column_names=("topological_ordering", "stream_ordering"),
+            from_token=attr.astuple(from_token) if from_token else None,
```

Comment:
kinda feel like this should indirect via a method on the token class, so as to avoid tying it to one particular impl, but I don't feel hugely strongly.